controllers/cloud.redhat.com/providers/job/impl.go

```diff
@@ -1,15 +1,45 @@
 package job
 
 import (
+	"context"
+	"fmt"
+
 	crd "github.com/RedHatInsights/clowder/apis/cloud.redhat.com/v1alpha1"
+	deployProvider "github.com/RedHatInsights/clowder/controllers/cloud.redhat.com/providers/deployment"
+	"github.com/RedHatInsights/clowder/controllers/cloud.redhat.com/utils"
 	batchv1 "k8s.io/api/batch/v1"
 	core "k8s.io/api/core/v1"
-
-	deployProvider "github.com/RedHatInsights/clowder/controllers/cloud.redhat.com/providers/deployment"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"k8s.io/apimachinery/pkg/types"
 )
 
+func (jp *jobProvider) makePreHookJob(job *crd.Job, app *crd.ClowdApp) error {
+	randomString := utils.RandStringLower(7)
+	nn := types.NamespacedName{
+		Name:      fmt.Sprintf("%s-%s", job.Name, randomString),
+		Namespace: app.Namespace,
+	}
+
+	j := batchv1.Job{}
+	if err := jp.Cache.Create(PreHookJob, nn, &j); err != nil {
+		return err
+	}
+
+	CreatePreHookJobResource(app, nn, job, &j)
+	if err := jp.Cache.Update(PreHookJob, &j); err != nil {
+		return err
+	}
+
+	app.ObjectMeta.Annotations["clowder/pre-hook-status"] = "pending"
```

Comment:
Can you elaborate on the choice between annotations and status fields here please?